### PR TITLE
Implement capture image step.

### PR DESCRIPTION
**Pull request by boumenot**
Part of the changes needed for #94

### CODE REVIEW

**Comment by boumenot** [comment]
This commit is behind the Azure Client PR.

File: packer/builder/azure/arm/step_capture_image.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		say:    func(message string) { ui.Say(message) },
+		error:  func(e error) { ui.Error(e.Error()) },
+	}
+
+	step.capture = step.captureImage
+	return step
+}
+
+func (s *StepCaptureImage) captureImage(resourceGroupName string, computeName string, parameters *compute.VirtualMachineCaptureParameters) error {
+	generalizeResponse, err := s.client.Generalize(resourceGroupName, computeName)
```
Comment (paulmey) [inline packer/builder/azure/arm/step_capture_image.go]: I believe that this only 'marks' it as generalized. We'll still need to execute the actual commands to generalize the VM.

File: packer/builder/azure/arm/step_capture_image.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	s.client.VirtualMachinesClient.PollAsNeeded(generalizeResponse.Response)
+
+	captureResponse, err := s.client.Capture(resourceGroupName, computeName, *parameters)
+	if err != nil {
+		return err
+	}
+
+	s.client.VirtualMachinesClient.PollAsNeeded(captureResponse.Response.Response)
```
Comment (paulmey) [inline packer/builder/azure/arm/step_capture_image.go]: can't this eventually yield an error that we want to surface?

File: packer/builder/azure/arm/step_capture_image.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+	err := s.capture(resourceGroupName, computeName, parameters)
+	if err != nil {
+		state.Put(constants.Error, err)
+		s.error(err)
+
+		return multistep.ActionHalt
+	}
+
+	state.Put(constants.ArmIsValid, true)
```
Comment (paulmey) [inline packer/builder/azure/arm/step_capture_image.go]: ? ArmIsValid suggests that something was validated, whereas this step just captured a VM?

**Comment by paulmey** [comment]
LGTM otherwise

File: packer/builder/azure/arm/step_capture_image.go (hunk 3 above)
Comment (boumenot) [inline packer/builder/azure/arm/step_capture_image.go]: You're right, this is me being stupid. :smile:.  I shall correct!

File: packer/builder/azure/arm/step_capture_image.go (hunk 1 above)
Comment (boumenot) [inline packer/builder/azure/arm/step_capture_image.go]: Correct.  There's a distinct build step to make it generalized - I reuse the code from existing builder.

File: packer/builder/azure/arm/step_capture_image.go (hunk 2 above)
Comment (boumenot) [inline packer/builder/azure/arm/step_capture_image.go]: It will be yielded in the Run method.

**Pull request merged**